E0631 in closure arguments suggest invalid type parameter

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code:

```rust
fn main() {
    "".starts_with(|ch| char::is_ascii(ch));
}
```

I expected to see this happen: *explanation*

```
error[E0631]: type mismatch in closure arguments
    --> src/bin/main.rs:2:8
     |
   2 |     "".starts_with(|ch| char::is_ascii(ch));
     |        ^^^^^^^^^^^ ---- found signature defined here
     |        |
     |        expected due to this
     |
     = note: expected closure signature `fn(_) -> _`
                found closure signature `fn(&_) -> _`
     = note: required for `{closure@src/bin/main.rs:2:20: 2:24}` to implement `Pattern`
note: required by a bound in `core::str::<impl str>::starts_with`
    --> /home/lrne/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/str/mod.rs:1378:27
     |
1378 |     pub fn starts_with<P: Pattern>(&self, pat: P) -> bool {
     |                           ^^^^^^^ required by this bound in `core::str::<impl str>::starts_with`
help: consider adjusting the signature so it does not borrow its argument
     |
   2 |     "".starts_with(|ch: char| char::is_ascii(ch));
     |                       ++++++

For more information about this error, try `rustc --explain E0631`.
error: could not compile `quick-rs` (bin "main") due to 1 previous error
```

Instead, this happened: (unexpected change on closure arguments)

```
error[E0631]: type mismatch in closure arguments
    --> src/bin/main.rs:2:8
     |
   2 |     "".starts_with(|ch| char::is_ascii(ch));
     |        ^^^^^^^^^^^ ---- found signature defined here
     |        |
     |        expected due to this
     |
     = note: expected closure signature `fn(_) -> _`
                found closure signature `fn(&_) -> _`
     = note: required for `{closure@src/bin/main.rs:2:20: 2:24}` to implement `Pattern`
note: required by a bound in `core::str::<impl str>::starts_with`
    --> /home/lrne/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/str/mod.rs:1378:27
     |
1378 |     pub fn starts_with<P: Pattern>(&self, pat: P) -> bool {
     |                           ^^^^^^^ required by this bound in `core::str::<impl str>::starts_with`
help: consider adjusting the signature so it does not borrow its argument
     |
   2 |     "".starts_with(|char| char::is_ascii(ch));
     |                       ++

For more information about this error, try `rustc --explain E0631`.
error: could not compile `quick-rs` (bin "main") due to 1 previous error
```

### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
rustc 1.93.0-nightly (f15a7f385 2025-11-04)
binary: rustc
commit-hash: f15a7f38580ddbdc1a23909dd05cf6cc6d9f3919
commit-date: 2025-11-04
host: x86_64-unknown-linux-gnu
release: 1.93.0-nightly
LLVM version: 21.1.3
```

@rustbot label +A-diagnostics

--- a/compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs
+++ b/compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs
@@ -5323,12 +5323,9 @@ fn hint_missing_borrow<'tcx>(
                     ty = mut_ty.ty;
                     left -= 1;
                 }
-                let sugg = if left == 0 {
-                    (span, String::new())
-                } else {
-                    (arg.span, expected_arg.to_string())
-                };
-                remove_borrow.push(sugg);
+                if left == 0 {
+                    remove_borrow.push((span, String::new()));
+                }
             }
         }
     }

--- a/tests/ui/closures/multiple-fn-bounds.stderr
+++ b/tests/ui/closures/multiple-fn-bounds.stderr
@@ -19,11 +19,6 @@ note: required by a bound in `foo`
    |
 LL | fn foo<F: Fn(&char) -> bool + Fn(char) -> bool>(f: F) {
    |                               ^^^^^^^^^^^^^^^^ required by this bound in `foo`
-help: consider adjusting the signature so it does not borrow its argument
-   |
-LL -     foo(move |x| v);
-LL +     foo(move |char| v);
-   |
 
 error: aborting due to 1 previous error
 

